### ISSUE TITLE
Invalid annotation failure for @receiver target annotation with parameter

### ISSUE THREAD

**Issue by adamsp** [pinterest/ktlint#885]
## Expected Behavior

`@receiver` annotations with parameters should not be required to be on a different line.

## Observed Behavior

When annotating with `@receiver:SomeAnnotation(SomeParameter)`, the [`AnnotationRule`](https://github.com/pinterest/ktlint/blob/4dda69b71fbbde261711b5424bf4846977648bac/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/AnnotationRule.kt) fails with the following:
> Annotations with parameters should all be placed on separate lines prior to the annotated construct

## Steps to Reproduce

Drop this in a Kotlin file and run ktlint on it:
```kotlin
interface AnnotationExample {
  fun @receiver:ExampleAnnotation("Breaks Ktlint") ExampleClass.provideExample(): ExampleInterface
}

interface ExampleInterface
class ExampleClass : ExampleInterface

annotation class ExampleAnnotation(val arg: String)
```

You can successfully meet the criteria with the following:

```kotlin
  fun @receiver:ExampleAnnotation("Breaks Ktlint")
  ExampleClass.provideExample(): ExampleInterface
```
However this is quite unexpected as a reader, and feels inconsistent when parameter-free annotations work just fine:

```kotlin
fun @receiver:ExampleAnnotation ExampleClass.provideExample(): ExampleInterface
```

## Your Environment

```
./gradlew ktlint --version

------------------------------------------------------------
Gradle 6.6.1
------------------------------------------------------------

Build time:   2020-08-25 16:29:12 UTC
Revision:     f2d1fb54a951d8b11d25748e4711bec8d128d7e3

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          1.8.0_242-release (JetBrains s.r.o 25.242-b3-6222593)
OS:           Mac OS X 10.15.6 x86_64
```

**Comment by romtsn** [comment]
Thanks for reporting! We'll take a look. And the version of ktlint is 0.37.2 I assume (it seems you've missed to add it to the report)?

**Comment by ZacSweers** [comment]
We're on 0.38.1
